### PR TITLE
Move webDriver to Page

### PR DESCRIPTION
**Pull request by 0I0u3uk**
Драйвер не статический сейчас.

Что еще нужно сделать:
+ Нужно пересмотреть структуру driverExtensions
+ сделать чтобы pageContext хранил данные для мультидрейдинга, и мог отдавать currenPage для текущего треда
+ убрать из PageFactory actions и пр что там быть не должно

### CODE REVIEW

File: java/plugins/web-elements-plugin/src/test/java/pagefactory/pages/webelements/IndexPage.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 package pagefactory.pages.webelements;
 
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import ru.sbtqa.tag.pagefactory.PageFactory;
```
Comment (varivoda) [inline java/plugins/web-elements-plugin/src/test/java/pagefactory/pages/webelements/IndexPage.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/IndexPage.java [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import com.epam.jdi.uitests.web.selenium.elements.common.Button;
 import com.epam.jdi.uitests.web.selenium.elements.common.Text;
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.FindBy;
 import ru.sbtqa.tag.pagefactory.JDIUtils;
```
Comment (varivoda) [inline java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/IndexPage.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.JDIUtils'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/StepDefs.java [hunk 3]
```diff
@@ -3,11 +3,12 @@
 import cucumber.api.java.Before;
 import ru.sbtqa.tag.pagefactory.JDIUtils;
 import ru.sbtqa.tag.pagefactory.PageFactory;
```
Comment (varivoda) [inline java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/StepDefs.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/StepDefs.java [hunk 4]
```diff
@@ -3,11 +3,12 @@
...
+import ru.sbtqa.tag.pagefactory.drivers.TagWebDriver;
 
 public class StepDefs {
     
     @Before
     public void initJDI() {
-        JDIUtils.setJDIConfig(() -> PageFactory.getDriver());
+        JDIUtils.setJDIConfig(() -> TagWebDriver.getDriver());
```
Comment (varivoda) [inline java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/StepDefs.java]: MINOR Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)

File: java/plugins/web-elements-plugin/src/test/java/pagefactory/pages/webelements/ContactPage.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package pagefactory.pages.webelements;
 
 import org.junit.Assert;
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import ru.sbtqa.tag.pagefactory.PageFactory;
```
Comment (varivoda) [inline java/plugins/web-elements-plugin/src/test/java/pagefactory/pages/webelements/ContactPage.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/ScreenShooter.java [hunk 6]
```diff
@@ -12,6 +12,7 @@
 import org.openqa.selenium.TakesScreenshot;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import ru.sbtqa.tag.pagefactory.PageContext;
 import ru.sbtqa.tag.pagefactory.PageFactory;
```
Comment (varivoda) [inline java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/support/ScreenShooter.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/plugins/html-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/htmlelements/IndexPages/IndexPage.java [hunk 7]
```diff
@@ -1,5 +1,6 @@
 package ru.sbtqa.tag.pagefactory.pages.htmlelements.IndexPages;
 
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.FindBy;
 import ru.sbtqa.tag.pagefactory.HTMLPage;
 import ru.sbtqa.tag.pagefactory.PageFactory;
```
Comment (varivoda) [inline java/plugins/html-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/htmlelements/IndexPages/IndexPage.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java [hunk 8]
```diff
@@ -8,6 +9,7 @@
 import org.openqa.selenium.WebElement;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import ru.sbtqa.tag.pagefactory.PageContext;
 import ru.sbtqa.tag.pagefactory.PageFactory;
```
Comment (varivoda) [inline java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/extensions/MobileExtension.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/plugins/web-elements-plugin/src/test/java/pagefactory/pages/webelements/AbstractPage.java [hunk 9]
```diff
@@ -1,5 +1,6 @@
 package pagefactory.pages.webelements;
 
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
 import ru.sbtqa.tag.pagefactory.PageFactory;
```
Comment (varivoda) [inline java/plugins/web-elements-plugin/src/test/java/pagefactory/pages/webelements/AbstractPage.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/ContactPage.java [hunk 10]
```diff
@@ -7,6 +7,7 @@
 import com.epam.jdi.uitests.web.selenium.elements.complex.RadioButtons;
 import com.epam.jdi.uitests.web.selenium.elements.complex.Selector;
 import org.junit.Assert;
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.support.FindBy;
 import ru.sbtqa.tag.pagefactory.JDIUtils;
```
Comment (varivoda) [inline java/plugins/jdi-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/jdielements/ContactPage.java]: MINOR Remove this unused import 'ru.sbtqa.tag.pagefactory.JDIUtils'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by varivoda** [comment]
SonarQube analysis reported 16 issues
* ![MINOR][MINOR] 16 minor

Watch the comments in this conversation to review them.

#### 6 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [PageFactory.java#L3](https://github.com/sbtqa/page-factory-2/blob/3db7a4920d0909fa1df75c5cba2ce9f2c4db1210/java/core/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/PageFactory.java#L3): Remove this unused import 'io.appium.java_client.AppiumDriver'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ContactPage.java#L8](https://github.com/sbtqa/page-factory-2/blob/3db7a4920d0909fa1df75c5cba2ce9f2c4db1210/java/plugins/html-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/htmlelements/ContactPages/ContactPage.java#L8): Remove this unused import 'ru.sbtqa.tag.pagefactory.PageFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ContactPage.java#L17](https://github.com/sbtqa/page-factory-2/blob/3db7a4920d0909fa1df75c5cba2ce9f2c4db1210/java/plugins/html-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/htmlelements/ContactPages/ContactPage.java#L17): Remove this unused import 'ru.yandex.qatools.htmlelements.loader.decorator.HtmlElementDecorator'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ContactPage.java#L18](https://github.com/sbtqa/page-factory-2/blob/3db7a4920d0909fa1df75c5cba2ce9f2c4db1210/java/plugins/html-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/htmlelements/ContactPages/ContactPage.java#L18): Remove this unused import 'ru.yandex.qatools.htmlelements.loader.decorator.HtmlElementLocatorFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [IndexPage.java#L11](https://github.com/sbtqa/page-factory-2/blob/3db7a4920d0909fa1df75c5cba2ce9f2c4db1210/java/plugins/html-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/htmlelements/IndexPages/IndexPage.java#L11): Remove this unused import 'ru.yandex.qatools.htmlelements.loader.decorator.HtmlElementDecorator'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [IndexPage.java#L12](https://github.com/sbtqa/page-factory-2/blob/3db7a4920d0909fa1df75c5cba2ce9f2c4db1210/java/plugins/html-elements-plugin/src/test/java/ru/sbtqa/tag/pagefactory/pages/htmlelements/IndexPages/IndexPage.java#L12): Remove this unused import 'ru.yandex.qatools.htmlelements.loader.decorator.HtmlElementLocatorFactory'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by varivoda** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by kosteman** [comment]
absorbed by #49

**Pull request closed without merging**